### PR TITLE
Chore improve eslint setup

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

refactor + bugfixes

**Did you add tests for your changes?**

No, existing tests

**If relevant, did you update the documentation?**

No need

**Summary**

- Simplify eslint configuraions
- Setup prettier for ts normally + format it
- Fix some bugs - we use packages but not listen them in package.json

**Does this PR introduce a breaking change?**

No

**Other information**

No

### CODE REVIEW

File: lint-staged.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    '*.{json,md,yml,css}': ['prettier --write'],
+    '*.{js,ts}': ['eslint --fix'],
```
Comment (anshumanv) [inline lint-staged.config.js:3]: Can't we use prettier to format js and ts too?

Comment (evilebottnawi) [reply lint-staged.config.js:3]: We use prettier as eslint plugin

Comment (anshumanv) [reply lint-staged.config.js:3]: Makes sense, thanks

**Comment by anshumanv** [comment]
Conflicts

**Comment by evilebottnawi** [comment]
Yep, I will fix it

**Comment by evilebottnawi** [comment]
What is hell with `eslint-plugin-node` :confused:

**Pull request merged**